modules/core/LightSky.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "ILight.h"
+#include "CosineSampler.h"
+namespace rt {
+	/**
+	 * @brief Point light source class
+	 */
+	class CLightSky : public ILight
+	{
+	public:
+		/**
+		 * @brief Constructor
+		 * @param intensity The emission color and strength of the light source
+		 * @param position The position (origin) of the light source
+		 * @param castShadow Flag indicatin whether the light source casts shadow
+		 */
+		DllExport CLightSky(Vec3f direction, float radius,size_t nSamples, bool castShadow = false)
```

Comment:
Please remove *direction*
Instead of *nSamples* perhaps it is possible to use a sampler class